tools/query_breakdown/src/main/java/com/google/bigquery/ReplacementLogic.java

```diff
@@ -9,23 +10,50 @@
 
   /**
    * Given a component, provides a recommendation as to which component the input should be
-   * replaced with. Returns a list of components that it recommends.
+   * replaced with. Returns a list of components that it recommends. The default method of
+   * choosing recommendations is through randomization.
    *
    * We choose ArrayLists as the data structure because we want to impose a certain ordering
    * with the recommendations: certain recommendations should be "better" than others
+   *
+   * n is an integer that controls the number of replacement options
    */
   public static ArrayList<String> replace (String component, ArrayList<String> options) {
-    // simply returns the first n options
     int n = 3;
 
+    ArrayList<String> result = new ArrayList<>();
+
     if (options.size() <= n) {
-      return new ArrayList<>(options);
+      for (int i = 0; i < options.size(); i++) {
+        // gets rid of cases such as <QUOTED_STRING>
+        if (options.get(i).charAt(0) != '<' || options.get(i).length() <= 1) {
+          result.add(options.get(i));
+        }
+      }
+      return result;
     }
 
-    ArrayList<String> result = new ArrayList<>();
-    for (int i = 0; i < n; i++) {
-      result.add(options.get(i));
+    HashSet<Integer> seen = new HashSet<>();
+    // randomly populate result until full
+    while (result.size() < n && seen.size() < options.size()) {
+      int random = randomNumber(options.size() - 1);
+      if (seen.contains(random)) {
+        continue;
+      }
+      else if (options.get(random).charAt(0) == '<' && options.get(random).length() > 1) {
+        seen.add(random);
+        continue;
+      }
+      else {
+        result.add(options.get(random));
+        seen.add(random);
+      }
     }
     return result;
   }
+
+  // helper function that returns a random integer between 0 - n
+  private static int randomNumber(int n) {
```

Comment:
You should use `Random.nextInt()` (https://docs.oracle.com/javase/8/docs/api/java/util/Random.html#nextInt--)